File: playframework-10590/app/controllers/HomeController.scala
package controllers

import javax.inject._
import play.api.mvc._

@Singleton
class HomeController @Inject()(val controllerComponents: ControllerComponents) extends BaseController {

  def index() = Action { implicit request: Request[AnyContent] =>
    val checkboxCount = request.getQueryString("checkboxCount").map(_.toInt).getOrElse(10000)
    Ok(views.html.index(checkboxCount))
  }

  def testForPost() = Action(parse.multipartFormData) { implicit request: Request[MultipartFormData[play.api.libs.Files.TemporaryFile]] =>
    Redirect(controllers.routes.HomeController.index())
  }
}

File: playframework-11496/app/controllers/HomeController.java
package controllers;

import play.libs.Files;
import play.mvc.*;

public class HomeController extends Controller {

    public Result index() {
        return ok("hello world");
    }

    public Result test(Http.Request request) {
        Http.MultipartFormData<Files.TemporaryFile> body=request.body().asMultipartFormData();
        return ok("test");
    }

}
